pkg/util/prefetch/reader.go

```diff
@@ -88,7 +100,12 @@ func (r *Reader) Read(data []byte) (int, error) {
 	}
 }
 
-// Close implements io.Closer.
+// Close implements io.Closer. Close should not be called concurrently with Read.
 func (r *Reader) Close() error {
-	return r.r.Close()
+	ret := r.r.Close()
+	r.closeOnce.Do(func() {
+		close(r.closed)
```

Comment:
we don't close reader twice, seems no need this closeOnce

Reply:
I simply checked s3.go. In some cases s3ObjectReader will be called by line 893 and line 911

https://github.com/pingcap/tidb/blob/3543275dcf4b6454eb874c1362c87d31a963da6d/br/pkg/storage/s3.go#L872-L905

So I think we should tolerate the caller call Close() for multiple times. But all reader methods are not thread safe, I will change it to a bool rather than sync.Once